src/Features/CSharp/Portable/Completion/CompletionProviders/ImportCompletion/TypeImportCompletionProvider.cs

```diff
@@ -54,5 +56,21 @@ protected override bool IsFinalSemicolonOfUsingOrExtern(SyntaxNode directive, Sy
                 _ => false,
             };
         }
+
+        protected override async Task<bool> ShouldProvideParenthesisCompletionAsync(Document document, int position, ISymbol? symbol, char? commitKey, CancellationToken cancellationToken)
+        {
+            if (commitKey == ';')
+            {
+                var syntaxTree = await document.GetRequiredSyntaxTreeAsync(cancellationToken).ConfigureAwait(false);
+                var preProcessorTokenOnLeftOfPosition = syntaxTree.FindTokenOnLeftOfPosition(position, cancellationToken, includeDirectives: true);
+                var isPreProcessorDirectiveContext = syntaxTree.IsPreProcessorDirectiveContext(position, preProcessorTokenOnLeftOfPosition, cancellationToken);
+                var leftToken = isPreProcessorDirectiveContext
```

Comment:
What is it doing in terms of preprocessor context? Add comments

Reply:
I just copy the code in CSharpSyntaxContext here (because we don't care about other syntax context info) But sure I will add comments here

Reply:
Actually I feel it might not need to check the Directive
```C#
void Main()
{
var a = new 
#if DEB
}
```
If I am typing 'DEBUG', I don't think we need any completion for unimported type here.